### DISCUSSION TITLE
[Bug] Set-PnPHubSite -EnablePermissionsSync not setting the right property

### DISCUSSION THREAD

**Discussion by brunostampfli** [pnp/powershell#4187] [General]
### Reporting an Issue or Missing Feature
I would like to enable / disable 'Sync hub permissions to this site'.
`Set-PnPHubSite -EnablePermissionsSync` should do this.

### Expected behavior 
I use:

`Set-PnPHubSite -Identity "https://tenant.sharepoint.com/sites/myhubsite" -EnablePermissionsSync`

and

`Set-PnPHubSite -Identity "https://tenant.sharepoint.com/sites/myhubsite" -EnablePermissionsSync:$false`

My expectation is these commands should set the 'Sync hub permission to this site' On or Off.

### Actual behavior 
With these commands I can hide or show the controls on the Site Information Page, but the sync setting is NOT set Off or On.

### Steps to reproduce behavior
Run:

`Set-PnPHubSite -Identity "https://tenant.sharepoint.com/sites/myhubsite" -EnablePermissionsSync:$false`

This hides the setting, but it should set set setting to Off.

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
I tested this with 2.9.0 and 1.12.0.

### Which operating system/environment are you running PnP PowerShell on?
- [ x ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by jackpoz** [comment]
Just a note that this command simply sets the CSOM property `EnablePermissionsSync` of `Microsoft.Online.SharePoint.TenantAdministration.HubSiteProperties` class to true/false. The behavior you experience should be the same as using CSOM directly.

This is the line that sets the property:
https://github.com/pnp/powershell/blob/d8955c4e01721e5f4a25c1e548969a66e5e1d82a/src/Commands/Admin/SetHubSite.cs#L69

**Comment by brunostampfli** [comment]
Ok. I have examined the setting more closely.

Set-PnPHubSite sets the property always on the hub site.

With Set-PnPHubSite -Identity “https://tenant.sharepoint.com/sites/myhubsite” -EnablePermissionsSync
I activate the synchronization of permissions from the hub site to the child site.

On the child site I can switch the option 'Sync hub permissions to this site' on and off.

Is it possible to set this switch with PnP PowerShell?

![image](https://github.com/user-attachments/assets/2e1ca94e-16b6-46f6-9f58-d8f2a22c1de0)

This issue is now a question and no longer a bug.

**Comment by gautamdsheth** [comment]
Thanks @brunostampfli for confirming , moving this to discussion.

AFAIK, there's no method available at the moment for this, but maybe someone in the community might have a better answer

**Comment by brunostampfli** [comment]
I found a solution myself:

```
Connect-PnPOnline -Url "site url"
$site = Get-PnPSite
$site.CanSyncHubSitePermissions = $false
Invoke-PnPQuery
```
